### ISSUE TITLE
Could not find module ‘Control.Comonad’

### ISSUE THREAD

**Issue by MonsieurCactus** [ekmett/semigroupoids#21]
Trying to install semigroupids... references some error in `comonad`

    src/Data/Functor/Bind.hs:91:8:
        Could not find module ‘Control.Comonad’
        Perhaps you haven't installed the "dyn" libraries for package ‘comonad-4.2.4’?
        Use -v to see a list of the files searched for.

    Failed to install semigroupoids-4.3
    cabal: Error: some packages failed to install:
    semigroupoids-4.3 failed during the building phase. The exception was:
    ExitFailure 1
    
Let's see... is comonad installed?

    me@ubuntu-14.04$ cabal install comonad
    Resolving dependencies...
    All the requested packages are already installed:
    comonad-4.2.4
    Use --reinstall if you want to reinstall anyway.

---

I removed comonad and reinstalled and got even more errors.

    Installing library in
    /home/jdm/.cabal/lib/x86_64-linux-ghc-7.8.4/comonad-4.2.4
    setup: Error: Could not find module: Control.Comonad with any suffix:
    ["dyn_hi"] in the search path: ["dist/build"]
    Failed to install comonad-4.2.4
    cabal: Error: some packages failed to install:
    comonad-4.2.4 failed during the final install step. The exception was:
    ExitFailure 1

This could be an issue with comonad rather than semigroupoids

**Comment by MonsieurCactus** [comment]
semigroupoids seems to install with `--disable-shared` https://github.com/ekmett/comonad/issues/9

    cabal install semigroupoids --disable-shared

Next, my iHaskell installation gets stuck on `profunctors` 

    me@ubuntu-14.04:~$ cabal install profunctors
    
    Loading package comonad-4.2.4 ... <command line>: can't load .so/.DLL for: libHScomonad-    
    4.2.4.so (libHScomonad-4.2.4.so: cannot open shared object file: No such file or directory)
    Failed to install profunctors-4.4.1

**Comment by ekmett** [comment]
Something is very very wrong with your install. You might try asking on #haskell on irc.freenode.net to see if someone has better ideas how to fix your install.

**Comment by MonsieurCactus** [comment]
Even after updating with `cabal install cabal-install` it still didn't recognize `sandbox init` and it fact `cabal -V` showed it was still version 16.

Using `where cabal` I found the faulty cabal executable from my `ghc-7.6` installation and replaced it with a symlink.  In my case

    sudo ln -s /home/jdm/.cabal/bin/cabal /usr/bin/cabal

So now I have cabal version 23, I opened a sandbox and `profunctors` installs perfectly.  Case closed.
